### ISSUE TITLE
Add support for MDX files

### ISSUE THREAD

**Issue by raulrpearson** [mdickin/vscode-markdown-shortcuts#46]
I'm using vscode-mdx to [add support for MDX](https://github.com/silvenon/vscode-mdx/blob/56840dbed0958f3efe8b505a22bb58d4d50eeda5/package.json#L24) in VS Code. I would like to use vscode-markdown-shortcuts on these MDX files. The workaround I'm using at the moment is to add this to my `settings.json` file:

```json
"files.associations": {
    "*.mdx": "markdown"
  }
```

But that defeats the purpose of having vscode-mdx installed 😅 .

Would it make sense for vscode-markdown-shortcuts to add support for MDX? It seems to me this could be done by changing all the `editorLangId == 'markdown'` in [`package.json`](https://github.com/mdickin/vscode-markdown-shortcuts/blob/master/package.json) to:

```json
(editorLangId == 'markdown' || editorLangId == 'mdx')
```

Would it be this easy? Would this be a desirable behaviour? Any side effects that I'm missing? Would it be better to put this behind a configuration flag?

I'm happy to put in a PR for this change.

**Comment by mdickin** [comment]
Hi Raul,

I'm worried about including "mdx" as a default supported langId, as some users may use it for something other than Markdown React. I know it's a database file type as well (though I doubt many people are opening those files in VSCode).

I would love to make it customizable, though. The extension already checks user settings to determine whether to show/hide menu items:

```
{
    "command": "md-shortcut.toggleBold",
    "when": "editorLangId == 'markdown' && config.markdownShortcuts.icons.bold",
    "group": "navigation@1"
}
```

I haven't dug into whether it's possible, but it would be great if we could provide a setting that contains an array of recognized languages and then do something like `"when": "config.markdownShortcuts.languages.indexOf(editorLangId) >= 0"` in all the places in `package.json` where we check the editorLangId.

If you want to submit a PR, I'd love to look it over. Let me know if you have any other implementation ideas.

**Comment by mdickin** [comment]
Actually, this might be the way forward instead of an array:

https://github.com/Microsoft/vscode/issues/22778#issuecomment-426711403

**Comment by raulrpearson** [comment]
So the aim would be to have a config option like this:

```json
"markdownShortcuts.languages": [
    "mdx",
    "mdy",
    "mdz"
]
```

Then transform that array into the string `/(markdown|mdx|mdy|mdz)/` and use that in the `"when"` logic in `package.json`?

One way to build that string would be to use this, which works even if the array is empty:

```javascript
languages.reduce((acc, elem) => acc + "|" + elem, "/(markdown") + ")/"
```

What I don't know is where that piece of code needs to go. I'm not familiar with VS Code extension development. Would an exported variable in [`extension.js`](https://github.com/mdickin/vscode-markdown-shortcuts/blob/master/extension.js) magically become available for use in `package.json`?

Would this be your preferred solution, @mdickin? If you can give me some pointers, I'll look into it and work on the pull request.

**Comment by mdickin** [comment]
So after looking over your last comment, I realized my approach wouldn't work. Even if we stored the regex as a configuration variable, I don't think it would recognize it as a regex like we need.

Instead, I think we can use a similar approach to what [gitlens](https://github.com/eamodio/vscode-gitlens) is doing with `setCommandContext`. On activation, we would do something like 

```
const languageArray = vscode.workspace.getConfiguration('markdownShortcuts').get('languages');
const languageRegex = new RegExp("(" + languageArray.join('|') + ")");
commands.executeCommand('setContext', 'markdownShortcuts:languages', languages);
```

Then the `when` clauses would look like `"when": "editorLangId =~ markdownShortcuts:languages"`

Remember that the default value of the new setting should _only_ have "markdown" in it.

Let me know if that doesn't make sense or if you run into any issues.
